FRCs/frc-00zz.md

```diff
@@ -0,0 +1,73 @@
+## Simple Summary
+Timelock encryption is a method of encrypting a message now that can't be decrypted until a specific point in the future has passed. The drand team have launched a scheme that uses BLS12-381 signatures in an innovative way using [identity based encryption](some-link-here) to achieve practical Timelock encryption using the drand network.
+As the Filecoin network uses keypairs compatible with BLS12-381, it could trivially implement built-in FVM actors for enabling timelock encryption (providing the [FIP for switching to the new drand network](some-link-here) is accepted).
+
+
+## Abstract
+On 2023-03-01, drand launched the first practical Timelock encryption scheme to mainnet after undergoing a security audit and developing libraries and tooling to make it easier.
+Timelock encryption would enable users to do sealed-bid auctions, create more secure implementations of [RANDAO](some-link-here), schedule social media network messages, prevent [MEV](some-link-here), and much, much more!
+The drand timelock scheme uses [identity based encryption](some-link-here) which exploits the pairing capabilities of BLS12-381 to use a future round number as a public key to encrypt messages which will be decryptable only once that future round has been emitted by the drand network by using the round signature as a private key.
+
+We propose a precompile for FVM that would enable users to timelock encrypt arbitrary data and attempt decryption of previously timelock encrypted data. This would enable users to timelock encrypt some part of the data inside their smart contract, while also enabling users to pass ciphertexts directly into smart contracts for decryption there at a later time. 
+
+## Change Motivation
+To enable timelock encryption and decryption using primitives that are already available in the FVM internals.
+
+## Specification
+
+The following interfaces should be fulfilled in FEVM and native actors respectively:
+
+```solidity
+interface Timelock {
+    function encrypt(bytes message, uint64 roundNumber) returns (bytes)
+    function decrypt(bytes message) returns (bytes, bool)
+}
```

Comment:
How do we account for the fact the chainhash and public key of the network might change in the future? Should it be parameters too?

Reply:
Hmm good question - probably don't want to push this decision onto the caller, but perhaps deploy different precompiles for different networks. I'm not sure if there's prior art on FVM for this kind of decision

Reply:
I've added a line about this in the section